client/my-sites/site-settings/analytics/form-google-analytics.jsx

```diff
@@ -88,8 +90,9 @@ export const GoogleAnalyticsForm = ( props ) => {
 		placeholderText,
 		recordSupportLinkClick,
 		setDisplayForm,
+		isFreeWPCOM,
 	};
-	if ( props.siteIsJetpack ) {
+	if ( props.siteIsJetpack && isGoogleAnalyticsEligible ) {
```

Comment:
I think `isGoogleAnalyticsEligible` returns `false` for Jetpack Free sites and that causes the regression I noted.